File: backend/app/api/repayments.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.database import get_db
from app.models.repayment import Repayment
from app.schemas.repayment import RepaymentCreate
from app.core.dependencies import get_current_user

router = APIRouter(
    prefix="/repayments",
    tags=["Repayments"]
)

@router.post("/")
def create_repayment(
    data: RepaymentCreate,
    current_user = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    repayment = Repayment(
        loan_id=data.loan_id,
        amount_paid=data.amount_paid,
        payment_date=data.payment_date,
        payment_method=data.payment_method,
        notes=data.notes
    )

    db.add(repayment)
    db.commit()
    db.refresh(repayment)

    return repayment

@router.get("/")
def get_repayments(
    current_user = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    return db.query(Repayment).all()

File: frontend-v2/src/lib/api/repayments.ts
import { apiFetch } from "./client";

export function getLoanRepayments(loanId: number)  
{ 
    return apiFetch( `/repayments/loan/${loanId}`
  );
}

export function createRepayment(data: {
  loan_id: number;
  amount_paid: number;
  payment_date: string;
  payment_method: string;
  notes?: string;
}) {
  return apiFetch("/repayments/", {
    method: "POST",
    body: JSON.stringify(data),
  });
}
